core/spec.md

```diff
@@ -2728,6 +2741,18 @@ The following describes the attributes of Registry model:
   - A value of `true` indicates that Resource of this type supports a separate
     document to be associated with it.
 
+- `groups.resources.singleversionroot`
+    - Type: Boolean (`true` or `false`, case-sensitive).
+    - OPTIONAL.
+    - Indicates whether Resources of this type can have multiple Versions
+      that have their `ancestor` attribute set to itself.
+    - When not specified, the default value MUST be `false`.
+    - A value of `true` indicates that only one Version of the Resource can
+      have its `ancestor` attribute set to itself. This is useful to avoid
+      creating multiple roots. When this attribute is set to `true`, the
+      server MUST reject the deletion of any Versions that would cause multiple
```

Comment:
server MUST generate an error if any request results in a
state where more than one Version of a Resource is a
root of an ancestor tree.
- - -
This will cover ALL cases of multiple roots being created, not just deletes.  E.g. an import that tries to create a 2nd one.

Reply:
Another situation that I think we need to mention in the spec, or at least the primer:
- singleversionroot=true
- maxversions limit is reached
- user tries to create a new version

The create needs to fail (assuming it'll delete the current root) because the net result is 2 roots/tree.